crates/modelardb_common/src/metadata/table_metadata_manager.rs

```diff
@@ -171,23 +177,26 @@ impl TableMetadataManager {
     /// Return the name of each table currently in the metadata Delta Lake. Note that this does not
     /// include model tables. If the table names cannot be retrieved, [`DeltaTableError`] is returned.
     pub async fn table_names(&self) -> Result<Vec<String>, DeltaTableError> {
-        self.table_names_of_type("table").await
+        self.table_names_of_type(TableType::Table).await
     }
 
     /// Return the name of each model table currently in the metadata Delta Lake. Note that this
     /// does not include tables. If the table names cannot be retrieved, [`DeltaTableError`] is
     /// returned.
     pub async fn model_table_names(&self) -> Result<Vec<String>, DeltaTableError> {
-        self.table_names_of_type("model_table").await
+        self.table_names_of_type(TableType::ModelTable).await
     }
 
-    // Return the name of tables of `table_type` where `table_type` is either "table" or
-    // "model_table". Returns [`DeltaTableError`] if `table_type` is not "table" or "model_table" or
-    // the table names cannot be retrieved.
-    async fn table_names_of_type(&self, table_type: &str) -> Result<Vec<String>, DeltaTableError> {
-        if table_type != "table" && table_type != "model_table" {
-            return Err(DeltaTableError::NoMetadata);
-        }
+    // Return the name of tables of `table_type` of `table_type`. Returns [`DeltaTableError`] if
+    // `table_type` is not "table" or "model_table" or the table names cannot be retrieved.
```

Comment:
I guess it is not necessary to mention that `table_type` can only be "table" or "model_table" now that the enum restricts it.

Reply:
Fixed.